File: examples/usage.py
import os
from pathlib import Path

import requests
from pydantic import BaseModel

BASE_URL = "https://your-api-url.com"
API_KEY = os.getenv("API_KEY", "")


class TokenCount(BaseModel):
    o200k_base: int
    cl100k_base: int


class ProcessDocumentResponse(BaseModel):
    markdown: str
    language: str
    mimetype: str
    token_count: TokenCount


def process_document(file_path: Path, use_llm: bool = False) -> ProcessDocumentResponse:
    if not file_path.exists():
        raise FileNotFoundError(f"File not found: {file_path}")

    with open(file_path, "rb") as file:
        files = {"file": (file_path.name, file)}
        data = {"use_llm": str(use_llm).lower()}

        response = requests.post(
            f"{BASE_URL}/process/document",
            headers={"X-API-Key": API_KEY},
            files=files,
            data=data,
        )
        response.raise_for_status()
        return ProcessDocumentResponse(**response.json())


def process_url(url: str) -> ProcessDocumentResponse:
    response = requests.post(
        f"{BASE_URL}/process/url", params={"url": url}, headers={"X-API-Key": API_KEY}
    )
    response.raise_for_status()
    return ProcessDocumentResponse(**response.json())

File: examples/usage.ts
import axios from "axios";
import FormData from "form-data";
import fs from "fs";
import { basename } from "path";

export const BASE_URL = "https://your-api-url.com";
export const API_KEY = process.env.API_KEY || "";

export interface TokenCount {
  o200k_base: number;
  cl100k_base: number;
}

export interface ProcessDocumentResponse {
  markdown: string;
  language: string;
  mimetype: string;
  token_count: TokenCount;
}

export async function processDocument(
  filePath: string,
  useLLM: boolean = false
): Promise<ProcessDocumentResponse> {
  if (!fs.existsSync(filePath)) {
    throw new Error(`File not found: ${filePath}`);
  }
  const formData = new FormData();
  formData.append("file", fs.createReadStream(filePath), basename(filePath));
  formData.append("use_llm", String(useLLM).toLowerCase());

  const response = await axios.post(`${BASE_URL}/process/document`, formData, {
    headers: {
      "X-API-Key": API_KEY,
      ...formData.getHeaders(),
    },
  });

  return response.data as ProcessDocumentResponse;
}

export async function processUrl(
  url: string
): Promise<ProcessDocumentResponse> {
  const response = await axios.post(`${BASE_URL}/process/url`, null, {
    params: { url },
    headers: {
      "X-API-Key": API_KEY,
    },
  });
  return response.data as ProcessDocumentResponse;
}
